src/EnvelopeForwarder.php

```diff
@@ -68,6 +75,16 @@ public function forward(Envelope $envelope): PromiseInterface
 
         $dsn = Dsn::createFromString($dsn);
 
+        $rateLimiter = $this->getRateLimiter($dsn);
+
+        // @TODO: We might need to get a little more advanced here and extract all items (and their headers) from the evelope and check each individually so we can remove the ones that are rate limited
+        $envelopeItemType = $envelope->getFirstItemType();
```

Comment:
We need to do this for profiles, as this is a dedicated rate-limit category and we currently emit them in the same envelope as the transaction.

Reply:
Ah, there we go, okay then we need to parse the full envelope, no problem! This makes sense to do anyway if we want to be a good citizen.